src/linker/Linker.Dataflow/FlowAnnotations.cs

```diff
@@ -228,15 +229,15 @@ TypeAnnotations BuildTypeAnnotations (TypeDefinition type)
 							}
 						} else if (methodMemberTypes != DynamicallyAccessedMemberTypes.None) {
 							_context.LogWarning (
-								$"The 'DynamicallyAccessedMembersAttribute' is not allowed on methods. It is allowed on method return value or method parameters though.",
-								2041, method, subcategory: MessageSubCategory.TrimAnalysis);
+								new DiagnosticString (DiagnosticId.DynamicallyAccessedMembersIsNotAllowedOnMethods).GetMessage (),
+								(int) DiagnosticId.DynamicallyAccessedMembersIsNotAllowedOnMethods, method, subcategory: MessageSubCategory.TrimAnalysis);
 						}
 					} else {
 						offset = 0;
 						if (methodMemberTypes != DynamicallyAccessedMemberTypes.None) {
 							_context.LogWarning (
```

Comment:
I wonder if we should add an overload to LogWarning which takes the enum instead of int - to avoid the casts in all the callsites.